File: packages/es-markdown-core/src/model/cross-link.ts
import type { MarkupBase } from "./base.js";
import type { Destination } from "../values.js";

/** A workspace link. The label is absent only when no separator was written. */
export interface CrossLink extends MarkupBase<"crossLink"> {
    readonly dest: Destination;
    /** The complete raw authored label; null only when no separator was authored. */
    readonly label: string | null;
}

File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/CrossLink.kt
package com.nouprax.markdown.core

/** A workspace link. [label] is absent only when no separator was written. */
public class CrossLink internal constructor(
    public val dest: Destination,
    /** The complete raw authored label. */
    public val label: String?,
    override val scope: Scope,
    override val anchor: String?,
    override val attributes: Attributes,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visit(this)
}
